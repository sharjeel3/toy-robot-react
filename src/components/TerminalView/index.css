.terminal {
    height: 100vh;
}
