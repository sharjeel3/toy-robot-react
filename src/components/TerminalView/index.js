import React from "react";
import Terminal from "terminal-in-react";
import "./index.css";
import { useToyRobotHook } from "../../hooks/ToyRobot";

const TerminalView = () => {
  const { getInstructionOutput } = useToyRobotHook();

  const commandFn = command => (args, print) => {
    /* Can do something fancy with error message in future
    Some commands do not return an output for print on screen */
    const [error, output] = getInstructionOutput({ command, args });
    if (error) {
      print(error);
    } else if (output) {
      print(output);
    }
  };

  return (
    <div className="terminal">
      <Terminal
        color="green"
        backgroundColor="black"
        barColor="black"
        startState="maximised"
        style={{ fontWeight: "bold", fontSize: "1em" }}
        commands={{
          PLACE: {
            method: commandFn("PLACE"),
            options: [{
              name: "place",
              defaultValue: undefined
            }]
          }
        }}
        descriptions={{
          show: false,
          PLACE: "PLACE toy robot at a given position(X,Y) and direction(F). Format is PLACE X,Y,F"
        }}
        msg="Welcome REA Team - This is a terminal emulator for Toy Robot simulator demo. Start typing below and press ENTER to run the command."
      />
    </div>
  );
};

export default TerminalView;
