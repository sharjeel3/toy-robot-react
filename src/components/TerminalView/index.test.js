import React from "react";
import renderer from "react-test-renderer";
import TerminalView from "./index";

it("should render terminal without crashing", () => {
  const tree = renderer.create(<TerminalView />).toJSON();
  expect(tree).toMatchSnapshot();
});
