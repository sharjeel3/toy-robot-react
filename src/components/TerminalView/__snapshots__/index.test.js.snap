// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render terminal without crashing 1`] = `
<div
  className="terminal"
>
  <div
    className="terminal-base sc-bdVaJa gSZAyM"
  >
    <div
      style={
        Object {
          "animation": "fadeIn 0.18s ease-in",
          "color": "green",
          "fontFamily": "'Inconsolata', monospace",
          "fontSize": "1em",
          "fontWeight": "bold",
          "height": "100%",
        }
      }
    >
      <div
        className="sc-bxivhb hdqhPL"
        style={
          Object {
            "maxWidth": "100%",
          }
        }
      >
        <svg
          height="20"
          width="100"
        >
          <circle
            cx="24"
            cy="14"
            fill="red"
            onClick={[Function]}
            r="5"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          />
          <circle
            cx="44"
            cy="14"
            fill="orange"
            onClick={[Function]}
            r="5"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          />
          <circle
            cx="64"
            cy="14"
            fill="green"
            onClick={[Function]}
            r="5"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          />
        </svg>
      </div>
      <div
        className="sc-jTzLTM lfnIny"
        style={
          Object {
            "maxWidth": "100%",
          }
        }
      >
        <div
          className="sc-cSHVUG iWWnLG"
          onClick={[Function]}
          onFocus={[Function]}
          tabIndex={0}
          title="bash"
        >
          bash
        </div>
        <div
          className="sc-fjdhpX ghCJNo"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="sc-kAzzGY hjGnyC"
            onClick={[Function]}
          >
            +
          </div>
        </div>
      </div>
      <div
        className="sc-EHOje dMFuoo"
        onKeyUp={[Function]}
        style={
          Object {
            "maxHeight": "calc(100% - 60px)",
            "maxWidth": "100%",
            "overflowY": "auto",
          }
        }
        tabIndex="0"
      >
        <div
          className="sc-gZMcBi cwkXTh"
        >
          <div
            className="sc-gqjmRU gAcuTz"
          >
            <div
              className="sc-VigVT bYwkqd"
            >
              <pre
                className="sc-gzVnrw duUAwb"
              >
                Welcome REA Team - This is a terminal emulator for Toy Robot simulator demo. Start typing below and press ENTER to run the command. 

You may start with 'help'
              </pre>
              <div
                className="sc-htoDjs iQSzyY"
              >
                <span
                  className="sc-dnqmqq dNrvp"
                >
                  &gt;
                </span>
                <input
                  className="sc-iwsKbI jLRkOo"
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  tabIndex="-1"
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
